tests/Fixer/FunctionNotation/PhpdocToParamTypeFixerTest.php

```diff
@@ -0,0 +1,290 @@
+                '<?php
+                    /**
+                    * @param int $foo
+                    * @param string $bar
+                    * @param Baz $hey
+                    * @param float $tab
+                    * @param bool $baz
+                    */
+                    function my_foo($bar, $foo, $baz, $tab, $hey) {}',
+            ],
+            'non-root class with massive string params' => [
+                '<?php
+                    /**
+                    * @param string $bar
+                    * @param string $baz
+                    * @param string $tab
+                    * @param string $foo
+                    */
+                    function my_foo(string $bar, string $baz, string $tab, string $foo) {}',
+                '<?php
+                    /**
+                    * @param string $bar
+                    * @param string $baz
+                    * @param string $tab
+                    * @param string $foo
+                    */
+                    function my_foo($bar, $baz, $tab, $foo) {}',
+            ],
+            'non-root class with different types of params' => [
+                '<?php
+                    /**
+                    * @param string $bar
+                    * @param int $baz
+                    * @param float $tab
+                    */
+                    function my_foo(string $bar, int $baz, float $tab) {}',
+                '<?php
+                    /**
+                    * @param string $bar
+                    * @param int $baz
+                    * @param float $tab
+                    */
+                    function my_foo($bar, $baz, $tab) {}',
+            ],
+            'non-root class with mixed type of param' => [
+                '<?php
+                    /**
+                    * @param mixed $bar
+                    */
+                    function my_foo($bar) {}',
+            ],
+            'non-root namespaced class' => [
+                '<?php /** @param My\Bar $foo */ function my_foo(My\Bar $foo) {}',
+                '<?php /** @param My\Bar $foo */ function my_foo($foo) {}',
+            ],
+            'root class' => [
+                '<?php /** @param \My\Bar $foo */ function my_foo(\My\Bar $foo) {}',
+                '<?php /** @param \My\Bar $foo */ function my_foo($foo) {}',
+            ],
+            'interface' => [
+                '<?php interface Foo { /** @param Bar $bar */ function my_foo(Bar $bar); }',
+                '<?php interface Foo { /** @param Bar $bar */ function my_foo($bar); }',
+            ],
+            'invalid void param on ^7.1' => [
+                '<?php /** @param null|void $foo */ function my_foo($foo) {}',
+            ],
+            'iterable return on ^7.1' => [
+                '<?php /** @param iterable $counter */ function my_foo(iterable $counter) {}',
+                '<?php /** @param iterable $counter */ function my_foo($counter) {}',
+                70100,
```

Comment:
You test `iterable` and `void`, thereby proving that the version-specific types work. I do wonder, do we want `object` tested as well, just because it has a different version?

Reply:
added an object test for 7.2